### PR TITLE
EVG-7947: shared button component with loading icon

### PR DESCRIPTION
**Pull request by tgrander**
### Shared button component

schedule-patch-loading

Adds some tasty features to the leafygreen button, like showing a loading indicator

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  dataCy,
+}) => {
+  return (
+    <LeafyGreenButton
+      data-cy={dataCy}
+      variant={mapVariantToLeafyGreenVariant[variant]}
+      onClick={onClick}
+      disabled={disabled}
+    >
+      {loading && <StyledIcon type="loading" />}
```
Comment (khelif96) [inline src/components/Button/index.tsx:38]: I think you can pass the Icon directly into the leafygreen button it has a `glyph` prop. 

https://github.com/mongodb/leafygreen-ui/blob/6b4cd8fecaa09350b739b9d3cf246a6ba30dfb3d/packages/button/src/Button.tsx#L298

Comment (tgrander) [reply src/components/Button/index.tsx:38]: leafygreen does not have this icon -- it's from antd

Comment (khelif96) [reply src/components/Button/index.tsx:38]: It takes a `ReactElement` as an input. So you should be able to pass `<StyledIcon/>` Directly into it.
https://github.com/mongodb/leafygreen-ui/blob/6b4cd8fecaa09350b739b9d3cf246a6ba30dfb3d/packages/button/src/Button.story.tsx#L39

Comment (tgrander) [reply src/components/Button/index.tsx:38]: oh neat

Comment (khelif96) [reply src/components/Button/index.tsx]: haha i had to do this exact thing like 100 times during my stitch rotation.

**Comment by khelif96** [comment]
> ### Shared button component
> schedule-patch-loading
> 
> Adds some tasty features to the leafygreen button, like showing a loading indicator

Do you think its possible to make the button not shift the surrounding buttons around it when the loading animation starts?

**Comment by tgrander** [comment]
> Do you think its possible to make the button not shift the surrounding buttons around it when the loading animation starts?

Not sure this is possible since the icon takes up space. What are the downsides of the adjacent content shifting 8px?

**Pull request merged**